hugr-model/src/v0/scope/link.rs

```diff
@@ -0,0 +1,113 @@
+use std::hash::{BuildHasherDefault, Hash};
+
+use fxhash::FxHasher;
+use indexmap::IndexSet;
+
+use crate::v0::{LinkIndex, RegionId};
+
+type FxIndexSet<K> = IndexSet<K, BuildHasherDefault<FxHasher>>;
+
+/// Table for tracking links between ports.
+///
+/// Two ports are connected when they share the same link. Links are named and
+/// scoped via isolated regions. The names of links must be unique within a
+/// single isolated region. Links from one isolated region are not visible in
+/// another. Links do not have a unique point of declaration.
+///
+/// # Examples
+///
+/// ```
+/// # pub use hugr_model::v0::RegionId;
+/// # pub use hugr_model::v0::scope::LinkTable;
+/// let mut links = LinkTable::new();
+/// links.enter(RegionId(0));
+/// let foo_0 = links.use_link("foo");
+/// let bar_0 = links.use_link("bar");
+/// assert_eq!(foo_0, links.use_link("foo"));
+/// assert_eq!(bar_0, links.use_link("bar"));
+/// let (num_links, num_ports) = links.exit();
+/// assert_eq!(num_links, 2);
+/// assert_eq!(num_ports, 4);
+/// ```
```

Comment:
Different links that are defined in different (closed) regions can have the same index. So the `assert_ne` would fail! A link is only uniquely identified by its index together with its closed region.